integration-tests/validator/src/test/java/org/apache/camel/quarkus/component/validator/it/ValidatorTest.java

```diff
@@ -16,84 +16,40 @@
  */
 package org.apache.camel.quarkus.component.validator.it;
 
+import io.quarkus.test.common.QuarkusTestResource;
 import io.quarkus.test.junit.QuarkusTest;
 import io.restassured.RestAssured;
 import io.restassured.http.ContentType;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
 
 @QuarkusTest
+@QuarkusTestResource(ValidatorTestResource.class)
 class ValidatorTest {
 
-    @Test
-    public void validXMLFromClassPath() {
+    @ParameterizedTest
+    @ValueSource(strings = { "classpath", "filesystem", "http" })
+    public void validXML(String scheme) {
 
         RestAssured.given()
                 .contentType(ContentType.XML)
                 .body("<message><firstName>MyFirstname</firstName><lastName>MyLastname</lastName></message>")
-                .post("/validator/classpath")
+                .post("/validator/validate/" + scheme)
                 .then()
                 .statusCode(200);
 
     }
 
-    @Test
-    public void invalidXMLFromClassPath() {
+    @ParameterizedTest
+    @ValueSource(strings = { "classpath", "filesystem", "http" })
+    public void inValidXML(String scheme) {
 
         RestAssured.given()
                 .contentType(ContentType.XML)
                 .body("<message><firstName>MyFirstname</firstName></message>")
-                .post("/validator/classpath")
+                .post("/validator/validate/" + scheme)
                 .then()
                 .statusCode(500);
```

Comment:
Here I wonder whether there is some body text (or any other way) from which we could see that what failed is really the XSD validation? 500 could mean a failure somewhere else in the stack, which we definitely would not want to accept as "test passed".

Reply:
@ppalaga I have added some checks based on the body, I think it will make more sense now, Thanks for pointing out !

Reply:
The tests are failing. You might want to forward the exception message to the client in `ValidatorResource`. Something like 

```
    public javax.ws.rs.core.Response processOrderInXml(String statement, @PathParam("scheme") String scheme) {
        try {
            return javax.ws.rs.core.Response.ok().entity(producerTemplate.requestBody("direct:" + scheme, statement, String.class));
        } catch (Exception e) {
                return javax.ws.rs.core.Response.serverError().entity(e.getMessage());
        }
            
    }
```

Reply:
> The tests are failing. You might want to forward the exception message to the client in `ValidatorResource`. Something like
> 
> ```
>     public javax.ws.rs.core.Response processOrderInXml(String statement, @PathParam("scheme") String scheme) {
>         try {
>             return javax.ws.rs.core.Response.ok().entity(producerTemplate.requestBody("direct:" + scheme, statement, String.class));
>         } catch (Exception e) {
>                 return javax.ws.rs.core.Response.serverError().entity(e.getMessage());
>         }
>             
>     }
> ```